src/riak_kv_qry_compiler.erl

```diff
@@ -2229,6 +2247,50 @@ two_element_key_range_cannot_match_test() ->
         compile(DDL, Q, 100)
     ).
 
+group_by_one_field_test() ->
```

Comment:
Update unit tests to cover all error cases as per the docs:
https://docs.google.com/document/d/1wkXdv2L9dMFp_YWf-C_MMMNhJZpScM9ud0AAUs4Oog0/edit

(apologies - these error and success cases were added to the spec after the code was written)